sdk/core/core-client-rest/CHANGELOG.md

```diff
@@ -1,15 +1,11 @@
 # Release History
 
-## 2.3.6 (Unreleased)
-
-### Features Added
-
-### Breaking Changes
-
-### Bugs Fixed
+## 2.4.0 (2025-05-01)
```

Comment:
Marking this as minor to reflect the scale of the refactor. Technically this could be a patch, but I don't have a strong opinion -- does anyone have a preference here? Practically don't think it will make much difference